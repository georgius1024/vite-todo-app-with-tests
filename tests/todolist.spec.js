import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import component from "../src/components/TodoList.vue";

describe("TodoList", () => {
  const items = ["Item 1", "Item 2", "Item 3"];
  let wrapper

  beforeEach(() => {
    wrapper = mount(component, { props: { items } });
  })

  it("mounts without errors", () => {
    expect(wrapper).toBeDefined();
    expect(wrapper.exists()).toBe(true);
  });
  it("renders list", () => {
    expect(wrapper.html()).toContain("Item 1");
    expect(wrapper.html()).toContain("Item 2");
    expect(wrapper.html()).toContain("Item 3");
  });
  it("renders all", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("emits events", () => {
    const removeControl = wrapper.find('[data-testid="item-remove-0"]');
    expect(removeControl.exists()).toBe(true);
    removeControl.trigger("click");
    expect(wrapper.emitted().remove).toHaveLength(1);
  });
});
