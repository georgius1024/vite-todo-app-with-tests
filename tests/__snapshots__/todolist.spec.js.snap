// Vitest Snapshot v1

exports[`TodoList > renders all 1`] = `
"<ul data-v-4afa86a5=\\"\\" class=\\"todo\\" data-testid=\\"item-list\\">
  <li data-v-4afa86a5=\\"\\" data-testid=\\"item-0\\"><span data-v-4afa86a5=\\"\\" class=\\"remove\\" data-testid=\\"item-remove-0\\">×</span> Item 1</li>
  <li data-v-4afa86a5=\\"\\" data-testid=\\"item-1\\"><span data-v-4afa86a5=\\"\\" class=\\"remove\\" data-testid=\\"item-remove-1\\">×</span> Item 2</li>
  <li data-v-4afa86a5=\\"\\" data-testid=\\"item-2\\"><span data-v-4afa86a5=\\"\\" class=\\"remove\\" data-testid=\\"item-remove-2\\">×</span> Item 3</li>
</ul>"
`;
